### PR TITLE
Prohibit unversioned python shebangs

### PR DESCRIPTION
**Pull request by mikhailnov**
CC @Conan-Kudo

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,10 +1,11 @@
 NAME = spec-helper
-VERSION = 0.31.46
+VERSION = 0.31.49
```
Comment (Conan-Kudo) [inline Makefile:2]: Why are we bumping two versions here?

**Review by Conan-Kudo — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (mikhailnov) [reply Makefile:2]: because the latest git tag is v0.31.48

Comment (Conan-Kudo) [reply Makefile:2]: :+1:

**Review by Conan-Kudo — APPROVED** [review]

**Review by berolinux — COMMENTED** [review]
I disagree with this... At least it shouldn't be enabled by default.
Everything can safely assume that /usr/bin/python is python 3.x, it has been that way for years in OMV now. No need to suddenly break something that has been that way forever and will stay that way until python 4 is released...

**Comment by Conan-Kudo** [comment]
Future proofing is generally a better strategy than causing painful breakages randomly throughout the stack. We've all learned this the hard way everywhere, and this is why even the dependency generator always makes versioned deps. We should always generate sane dependencies, and unversioned interpreter in packages is not sane.

**Comment by mikhailnov** [comment]
For some background. I have made this for usage in ROSA and sent back to OMV as spec-helper upstream, so for me personally it is mostly indifferent if you keep it enabled in OMV or not.

**Comment by mikhailnov** [comment]
However it may be easier if specs are in sync and if there are no packages named `python-*` with python3 stuff inside in OMV.

**Comment by berolinux** [comment]
It's 2 different approaches trying to accomplish 2 different things. There's reasons for and against either.

What happens when python 4 comes out?

Approach 1 (unversioned shebangs, /usr/bin/python = current version, python-* packages = current version, versioned python2-* packages etc. are for previous versions): Once /usr/bin/python points at python 4, python-* packages will update to python 4.x as well (may need a little sed magic to replace BuildRequires: pkgconfig(python3) with BuildRequires: pkgconfig(python4) or so). Thanks to the bytecompile BRPs, stuff that isn't compatible with python 4.x will fail at build time rather than at runtime and we'll see what needs to be fixed.
User systems will update e.g. python-sphinx for python 3.x to python-sphinx for python 4.x even if we introduce a python3-sphinx package for stuff that still requires python 3.x.
Goal of approach 1: Kill off python 3.x as soon as possible when 4.x is ready.

Approach 2 (forced versioned shebangs --> /usr/bin/python is irrelevant, version hardcoded in package names): Once /usr/bin/python points at python 4, everything will still use python 3 because everything calls python3 rather than python.
No clean way to ask a box with an old installation to move from python3-sphinx to python4-sphinx still exists, given python4-sphinx can't really obsolete python3-sphinx as long as the latter is still needed.
Goal of approach 2: Stay on python 3.x forever, update to 4.x only with manual intervention.


Obviously for an LTS minded system, approach 2 is better -- but for something fast moving that doesn't like to carry around compat packages forever, approach 1 has many advantages.

Given the python guys have learned from the mess that is the python2 -> python3 transition, chances are python4 will try harder than to be mostly compatible (and, as we're seeing already, introducing incompatible changes with deprecation warnings in the 3.x line), and approach 1 may well work without breaking too many things.

Of course they may also decide to make Python 3 the Python equivalent of Windows 10 (no more new version numbers ever) and we may see python 3.2000.0 in couple of centuries. In that case there's no difference that matters between approaches 1 and 2 (but 1 leads to nicer package names).

**Comment by mikhailnov** [comment]
05.04.2020 18:44, Bernhard Rosenkraenzer пишет:
>
> Approach 2 (forced versioned shebangs --> /usr/bin/python is irrelevant, version hardcoded in package names): Once /usr/bin/python points at python 4, everything will still use python 3 because everything calls python3 rather than python.
> No clean way to ask a box with an old installation to move from python3-sphinx to python4-sphinx still exists, given python4-sphinx can't really obsolete python3-sphinx as long as the latter is still needed.
> Goal of approach 2: Stay on python 3.x forever, update to 4.x only with manual intervention.
>
Things will break untill _all_ dependecies become available for python4 and it is unlikely for most cases

**Pull request merged**